### DISCUSSION TITLE
Translating `(+) <$> (*2) <*> (+10)`

### DISCUSSION THREAD

**Discussion by THOUSAND-SKY** [gcanti/fp-ts#1801] [Q&A] [answered]
`(+) <$> [3] <*> [4] -- => [7]` is `pure (+) <*> [3] <*> [4]` 
and is translated to `pipe([sum], array.ap([3]), array.ap([4])) // => [7]`.

But what is `(+) <$> (*2) <*> (+10)` translated to? What is `pure (+)` in this context?

**Comment by samhh** [comment, marked as answer]
I think that example relies on the applicative instance for `(->)`. [fp-ts-std](https://github.com/samhh/fp-ts-std) defines these instances, for which the following appears to behave equivalently:

```ts
import { pipe } from 'fp-ts/function'
import * as Fn from 'fp-ts-std/Function'
import * as Num from 'fp-ts-std/Number'

export const f = pipe(
  Fn.of(Num.add),
  Fn.ap(Num.multiply(2)),
  Fn.ap(Num.add(10)),
)
```

Here `pure (+)` translates to `Fn.of(Num.add)`.

**Reply by THOUSAND-SKY** [reply]
Very nice. Thank you.
